### ISSUE TITLE
Renaming a azurerm_api_management_api `operation_id` causing a destroy and create creates a race condition

### ISSUE THREAD

**Issue by stusklinar** [hashicorp/terraform-provider-azurerm#26261]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.8.5

### AzureRM Provider Version

v3.100.0

### Affected Resource(s)/Data Source(s)

azurerm_api_management_api

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_api_management" "example" {
  name                = "example-apim"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  publisher_name      = "My Company"
  publisher_email     = "company@terraform.io"

  sku_name = "Developer_1"
}

resource "azurerm_api_management_api" "example" {
  name                = "example-api"
  resource_group_name = azurerm_resource_group.example.name
  api_management_name = azurerm_api_management.example.name
  revision            = "1"
  display_name        = "Example API"
  path                = "example"
  protocols           = ["https"]

}


resource "azurerm_api_management_api_operation" "example" {
  operation_id        = "user-delete"
  api_name            = azurerm_api_management_api.example.name
  api_management_name = data.azurerm_api_management_api.example.api_management_name
  resource_group_name = data.azurerm_api_management_api.example.resource_group_name
  display_name        = "Delete User Operation"
  method              = "DELETE"
  url_template        = "/users/{id}/delete"
  description         = "This can only be done by the logged in user."

  template_parameter {
    name     = "id"
    type     = "number"
    required = true
  }

  response {
    status_code = 200
  }
}
```

### Debug Output/Panic Output

```shell
module.users_api.azurerm_api_management_api_operation.operation["delete-user"]: Destroying... [id=/subscriptions/----/resourceGroups/example-resources/providers/Microsoft.ApiManagement/service/example-apim/apis/users/operations/delete-user]
module.users_api.azurerm_api_management_api_operation.operation["user-delete"]: Creating...


Error: creating/updating Operation (Subscription: "----"
│ Resource Group Name: "example-resources"
│ Service Name: "example-apim"
│ Api: "users"
│ Operation: "user-delete"): unexpected status 400 (400 Bad Request) with error: ValidationError: One or more fields contain incorrect values:
│ 
│   with module.drivers_api.azurerm_api_management_api_operation.operation["user-delete"],
│   on .terraform/modules/_api/azurerm_api_management_api/main.tf line 24, in resource "azurerm_api_management_api_operation" "operation":
│   24: resource "azurerm_api_management_api_operation" "operation" {

```

### Expected Behaviour

Destroy should happen - and only then should the create start.

### Actual Behaviour

Destory and create happen in parallel causing Azure APIs to error

### Steps to Reproduce

Using the provided code above, `apply` initially - then change the `operation_id` to something different - ensuring the URL remains the same - duplicate URLs against different APIs are what causes this.

Once the `operation_id` has changed, re-run `terraform apply`, you should now get a race condition and an error.

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against v3.x of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
